Mathlib/Analysis/Discrete/MarcinkiewiczZygmund.lean

```diff
@@ -0,0 +1,290 @@
+      ∑ a in B, ∑ b in B, (∑ i, (f (a i) - f (b i))) ^ (2 * m) :=
+    fun ε hε ↦ step_two_aux A f _ hε fun z : n → ℝ ↦ univ.sum z ^ (2 * m)
+  rw [Finset.sum_congr rfl this, sum_const, card_piFinset, card_pair, prod_const, Finset.card_univ,
+    nsmul_eq_mul, Nat.cast_pow, Nat.cast_two, inv_pow, inv_mul_cancel_left₀]
+  · positivity
+  · norm_num
+
+private lemma step_three (f : α → ℝ) :
+    ∑ ε in ({-1, 1} : Finset ℝ)^^n,
+      ∑ a in A ^^ n, ∑ b in A ^^ n, (∑ i, ε i * (f (a i) - f (b i))) ^ (2 * m) =
+      ∑ a in A ^^ n, ∑ b in A ^^ n, ∑ k in piAntidiag univ (2 * m),
+          (multinomial univ k * ∏ t, (f (a t) - f (b t)) ^ k t) *
+            ∑ ε in ({-1, 1} : Finset ℝ)^^n, ∏ t, ε t ^ k t := by
+  simp only [@sum_comm _ _ (n → ℝ) _ _ (A ^^ n), ← Complex.abs_pow, sum_pow_eq_sum_piAntidiag]
+  refine sum_congr rfl fun a _ ↦ ?_
+  refine sum_congr rfl fun b _ ↦ ?_
+  simp only [mul_pow, prod_mul_distrib, @sum_comm _ _ (n → ℝ), ← mul_sum, ← sum_mul]
+  refine sum_congr rfl fun k _ ↦ ?_
+  rw [← mul_assoc, mul_right_comm]
+
+private lemma step_four {k : n → ℕ} :
+    ∑ ε in ({-1, 1} : Finset ℝ)^^n, ∏ t, ε t ^ k t = 2 ^ card n * ite (∀ i, Even (k i)) 1 0 := by
+  norm_num [sum_prod_piFinset _ fun i ↦ (· ^ k i), neg_one_pow_eq_ite, ite_add,
+    Fintype.prod_ite_zero]
+
+private lemma step_six {f : α → ℝ} {a b : n → α} :
+    ∑ k : n → ℕ in piAntidiag univ m,
+        (multinomial univ fun a ↦ 2 * k a : ℝ) * ∏ i, (f (a i) - f (b i)) ^ (2 * k i) ≤
+      m ^ m * (∑ i, (f (a i) - f (b i)) ^ 2) ^ m := by
+  simp only [sum_pow_eq_sum_piAntidiag, mul_sum, ← mul_assoc, pow_mul]
+  gcongr with k hk
+  rw [mem_piAntidiag] at hk
+  exact mod_cast multinomial_two_mul_le_mul_multinomial.trans (by rw [hk.1])
+
+private lemma end_step {f : α → ℝ} (hm : 1 ≤ m) (hA : A.Nonempty) :
+    (∑ a in A ^^ n, ∑ b in A ^^ n, ∑ k in piAntidiag univ m,
+      ↑(multinomial univ fun i ↦ 2 * k i) * ∏ t, (f (a t) - f (b t)) ^ (2 * k t)) / A.card ^ card n
+        ≤ (4 * m) ^ m * ∑ a in A ^^ n, (∑ i, f (a i) ^ 2) ^ m := by
+  let B := A ^^ n
+  calc
+    (∑ a in B, ∑ b in B, ∑ k : n → ℕ in piAntidiag univ m,
+      (multinomial univ fun i ↦ 2 * k i : ℝ) * ∏ t, (f (a t) - f (b t)) ^ (2 * k t)) /
+        A.card ^ card n
+    _ ≤ (∑ a in B, ∑ b in B, m ^ m * 2 ^ (m + (m - 1)) *
+          ((∑ i, f (a i) ^ 2) ^ m + (∑ i, f (b i) ^ 2) ^ m) : ℝ) / A.card ^ card n := by
+      gcongr; exact step_six.trans <| step_seven.trans step_eight
+    _ = _ := by
+      simp only [mul_add, sum_add_distrib, sum_const, nsmul_eq_mul, ← mul_sum]
+      rw [← mul_add, ← mul_add, ← two_mul, card_piFinset, prod_const, Finset.card_univ,
+        Nat.cast_pow, mul_div_cancel_left₀ _ (by positivity), ← mul_assoc, mul_assoc _ _ 2,
+        mul_comm 4, mul_pow, ← pow_succ, add_assoc, Nat.sub_add_cancel hm, ← two_mul, pow_mul]
+      norm_num
+
+namespace Real
+
+/-- The **Marcinkiewicz-Zygmund inequality** for real-valued functions, with a slightly better
```

Comment:
Can you spell out what the inequality says, in layman terms -- in particular since there is no explicit independence in the statement, while it's clearly the intuition to have in mind? Same thing for the other main theorems.